### PR TITLE
Update to Minio Azure Gateway

### PR DESCRIPTION
**Pull request by jamesneimanconsulting**
initial edits

- [X] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by deekoder** [comment]
adding @krishnasrinivas  who worked on azure gateway

File: docs/gateway/azure.md [hunk 1]
```diff
@@ -1,54 +1,83 @@
...
+
+```
 [2017-02-22 01:50:43 PST]     0B ferenginar/
 [2017-02-26 21:43:51 PST]     0B my-container/
 [2017-02-26 22:10:11 PST]     0B test-container1/
 ```
 
-### Known limitations
-Gateway inherits the following Azure limitations:
+### 3.3 Known Limitations
```
Comment (krishnasrinivas) [inline docs/gateway/azure.md]: Can you put `Known Limitations` section at one level higher? i.e it does not belong under `mc`

File: docs/gateway/azure.md [hunk 2]
```diff
@@ -1,54 +1,83 @@
...
-Minio Gateway adds Amazon S3 compatibility to Microsoft Azure Blob Storage.
 
-## Run Minio Gateway for Microsoft Azure Blob Storage
-### Using Docker
-```
+This topic describes how to use Minio Gateway to add Amazon S3 compatibility to Microsoft Azure Blob Storage.
+
+1. [Run Minio Gateway for Microsoft Azure Blob Storage](#run-minio-gateway-for-azure) 
+2. [Test Using Minio Browser](#test-using-minio-browser) 
+3. [Test Using Minio Client](#test-using-minio-client)
```
Comment (krishnasrinivas) [inline docs/gateway/azure.md]: You could add "known limitations" and "explore further" to the above list

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

File: docs/gateway/azure.md [hunk 3]
```diff
@@ -1,54 +1,84 @@
...
+## <a name="known-limitations"></a>4. Known Limitations
+
+Minio Gateway has the following limitations when used with Azure:
 
-- Only read-only bucket policy supported at bucket level, all other variations will return API Notimplemented error.
-- Bucket names with "." in the bucket name are not supported.
-- Non-empty buckets get removed on a DeleteBucket() call.
-- _List Multipart Uploads_ always returns empty list.
+- It only supports read-only bucket policies at the bucket level; all other variations will return `API Notimplemented`.
+- Bucket names cannot contain the period character.
```
Comment (sgandon) [inline docs/gateway/azure.md]: I would keep the "." for more clarity for non good English speakers.

File: docs/gateway/azure.md [hunk 4]
```diff
@@ -1,54 +1,84 @@
...
-Gateway inherits the following Azure limitations:
+## <a name="known-limitations"></a>4. Known Limitations
+
+Minio Gateway has the following limitations when used with Azure:
 
-- Only read-only bucket policy supported at bucket level, all other variations will return API Notimplemented error.
-- Bucket names with "." in the bucket name are not supported.
-- Non-empty buckets get removed on a DeleteBucket() call.
-- _List Multipart Uploads_ always returns empty list.
+- It only supports read-only bucket policies at the bucket level; all other variations will return `API Notimplemented`.
```
Comment (sgandon) [inline docs/gateway/azure.md]: Could you elaborate on this limitation cause I don't understand what it means in terms of API ? can you do a PUT object ?

File: docs/gateway/azure.md [hunk 5]
```diff
@@ -1,54 +1,84 @@
...
  -e "MINIO_SECRET_KEY=azurestorageaccountkey" \
  minio/minio gateway azure
 ```
 
-### Using Binary
-```
+### 1.2 Run Minio Gateway Using the Minio Binary
+
+```sh
 export MINIO_ACCESS_KEY=azureaccountname
```
Comment (sgandon) [inline docs/gateway/azure.md]:
```suggestion
export MINIO_ACCESS_KEY= azurestorageaccountname
```

File: docs/gateway/azure.md [hunk 6]
```diff
@@ -1,54 +1,84 @@
...
 export MINIO_SECRET_KEY=azureaccountkey
```
Comment (sgandon) [inline docs/gateway/azure.md]:
```suggestion
export MINIO_SECRET_KEY= azurestorageaccountkey
```

File: docs/gateway/azure.md [hunk 7]
```diff
@@ -1,54 +1,84 @@
...
+`mc` supports file systems and Amazon S3-compatible cloud storage services (AWS Signature v2 and v4).
+
+Test `mc` with the gateways using the instructions in the [Minio Client Quickstart Guide](https://docs.minio.io/docs/minio-client-quickstart-guide).
+
+### 3.1 Configure the Gateway Using Minio Client
+
+Use the following command to configure the gateway:
+
+```sh
 mc config host add myazure http://gateway-ip:9000 azureaccountname azureaccountkey
```
Comment (sgandon) [inline docs/gateway/azure.md]: here is this azureaccountname or azurestorageaccountname ?
and the same for azureaccountkey

File: docs/gateway/azure.md (hunk 3 above)
Comment (sgandon) [inline docs/gateway/azure.md]:
```suggestion
- Bucket names cannot contain the period (".") character.
```

**Comment by sgandon** [comment]
I also think it would be nice to have a chapter on using helm for deployment.

File: docs/gateway/azure.md (hunk 4 above)
Comment (jamesneimanconsulting) [reply docs/gateway/azure.md]: @sgandon , @eco-minio Please provide this detail.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

File: docs/gateway/azure.md (hunk 4 above)
Comment (sgandon) [reply docs/gateway/azure.md]: @jamesneimanconsulting I am not sure to understand your comment above ?

**Pull request merged**